python/services/metadata_service.py

```diff
@@ -0,0 +1,109 @@
+import requests
+import logging
+
+
+class MetadataService:
+    def __init__(self, api_url: str, api_token: str):
+        self.api_url = api_url
+        self.api_token = api_token
+
+    def get_acceptable_slack_users(self):
+        return [
+            {"username": "sam.pepper"},
+            {"username": "connor.glynn"}
+        ]
+
+    def get_acceptable_github_usernames(self):
+        return [
+            {"username": "PepperMoJ"},
+            {"username": "connormaglynn"}
+        ]
+
+    def get_existing_slack_users(self):
+        try:
+            response = requests.get(
+                f"{self.api_url}/slack_users", headers={"Authorization": f"Bearer {self.api_token}"})
+            if response.status_code == 200:
+                return response.json()
+            else:
+                logging.error(
+                    "Error fetching existing Slack usernames: %s", response.content)
+                return []
+        except Exception as e:
+            logging.error(
+                "An error occurred while fetching existing Slack usernames: %s", str(e))
+            return []
+
+    def filter_usernames(self, username_list: list[dict], accepted_username_list: list[dict]):
+        """Filter out all usernames deemed not acceptable.
+
+        Parameters:
+            username_list: Initial list containing all usernames
+            accepted_username_list: A list of acceptable usernames to include in the final list
+
+        Returns:
+            filtered_usernames: A list of filtered usernames
+        """
+
+        accepted_usernames_set = {user["username"]
+                                  for user in accepted_username_list}
+
+        filtered_usernames = [
+            user for user in username_list if user["username"] in accepted_usernames_set
+        ]
+
+        return filtered_usernames
+
+    def combine_user_data(self, slack_user_data, github_user_data):

```

Comment:
Type annotations missing here - probably be good to document decisions like adding type annotations in some sort of Code Practices document 🤔 Separate thing though 🎃